Sources/performance-worker/WorkerService.swift

```diff
@@ -87,7 +89,41 @@ final class WorkerService: Grpc_Testing_WorkerService.ServiceProtocol, Sendable
   ) async throws
     -> GRPCCore.ServerResponse.Stream<Grpc_Testing_WorkerService.Method.RunServer.Output>
   {
-    throw RPCError(code: .unimplemented, message: "This RPC has not been implemented yet.")
+    return ServerResponse.Stream { writer in
+      for try await message in request.messages {
+        switch message.argtype {
+        case let .some(.setup(serverConfig)):
+          try await self.setupServer(serverConfig)
+          try await withThrowingTaskGroup(of: Void.self) { group in
+            group.addTask {
+              let role = self.state.withLockedValue({ $0.role })
+              switch role {
+              case let .server(server):
+                try await server.run()
+              default:
+                ()
+              }
+            }
+            try await group.next()
+            group.cancelAll()
```

Comment:
No point cancelling all: there's only one task and we just waited for it to return on the previous line. At this point there's no work to cancel.